src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java

```diff
@@ -125,16 +124,20 @@ protected void configure(HttpSecurity http) throws Exception {
         // @formatter:off
         http
             .csrf()
-            .csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse())
-            .ignoringAntMatchers("/websocket/**").ignoringAntMatchers("/api/lti/launch/*")
-        .and()
+            .disable()
             .addFilterBefore(corsFilter, UsernamePasswordAuthenticationFilter.class)
-            .exceptionHandling().authenticationEntryPoint(problemSupport)
-            .accessDeniedHandler(problemSupport)
+                .exceptionHandling().authenticationEntryPoint(problemSupport)
+                .accessDeniedHandler(problemSupport)
         .and()
             .headers()
+            .contentSecurityPolicy("default-src 'self'; frame-src 'self' data:; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; img-src * data:; font-src 'self' data:")
```

Comment:
Extend policy to whitelist youtube:
```suggestion
            .contentSecurityPolicy("default-src 'self'; frame-src 'self' data: https://www.youtube-nocookie.com/embed/ https://www.youtube.com/embed/ https://youtu.be/embed/ https://youtube-nocookie.com/embed/ https://youtube.com/embed/; script-src 'self' 'unsafe-inline' 'unsafe-eval' https://www.youtube.com/iframe_api https://*.ytimg.com/; style-src 'self' 'unsafe-inline' https://*.ytimg.com/; img-src * data: https://*.youtube-nocookie.com https://*.youtube.com https://*.ytimg.com/ ; font-src 'self' data:; child-src 'self' https://www.youtube-nocookie.com/ https://www.youtube.com/ https://youtu.be/ https://youtube-nocookie.com/ https://youtube.com/; object-src 'self' https://*.youtube-nocookie.com https://*.youtube.com; connect-src 'self' https://www.youtube-nocookie.com/ https://www.youtube.com/ https://youtube-nocookie.com/ https://youtube.com/; media-src 'self' https://*.youtube-nocookie.com https://*.youtube.com")
```

Reply:
The problem is that instructors could also decide to use videos on other platforms, e.g. Vimeo or even custom ones from their university, so we should probably allow anything here